packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -184,21 +232,22 @@ export class PreferencesController extends BaseControllerV1<
    * @param addresses - List of addresses to use to generate new identities.
    */
   addIdentities(addresses: string[]) {
-    const { identities } = this.state;
-    addresses.forEach((address) => {
-      address = toChecksumHexAddress(address);
-      if (identities[address]) {
-        return;
+    const checksummedAddresses = addresses.map(toChecksumHexAddress);
+    this.update((state) => {
+      const { identities } = state;
+      for (const address of checksummedAddresses) {
+        if (identities[address]) {
+          return;
```

Comment:
Whoops, this should be a `continue` not a `return`. I'll follow up to add a test to catch this mistake as well.

Reply:
Fixed here: https://github.com/MetaMask/core/pull/3713/commits/b92b3796683fe625304ccecd65e66764fc7b72c6